File: 0005.LongestPalindromicSubstring/centerExpansion.cs
public class Solution {
    public string LongestPalindrome(string s) {
        int len = s.Length, max = 0;
        string ret = "";
        for(int i = 0; i < len; ++i){
            string s1 = getPalindromic(s, i, i);
            string s2 = getPalindromic(s, i, i + 1);
            if(s1.Length > max){
                ret = s1;
                max = s1.Length;
            }
            if(s2.Length > max){
                ret = s2;
                max = s2.Length;
            }
        }
        return ret;
    }
    string getPalindromic(string s, int low, int high){
        int len = s.Length;
        while(low >= 0 && high <= len - 1){
            char a = s[low], b = s[high];
            if(a == b){
                low--;high++;
            }else
                break;
        }
        return s.Substring(low + 1, high - low - 1);
    }
}

File: 0005.LongestPalindromicSubstring/centerExpansion.java
class Solution {
    public String longestPalindrome(String s) {
        int len = s.length();
        String ret = "";
        for(int i = 0; i < len; ++i){
            String s1 = getPalindrome(s, i, i);
            String s2 = getPalindrome(s, i, i + 1);
            if(s1.length() > ret.length()){
                ret = s1;
            }
            if(s2.length() > ret.length()){
                ret = s2;
            }
        }
        return ret;
    }

    private String getPalindrome(String s, int low, int high){
        int len = s.length();
        while(low >= 0 && high <= len - 1){
            char c1 = s.charAt(low), c2 = s.charAt(high);
            if(c1 == c2){
                low--;
                high++;
            }else{
                break;
            }
        }
        //public String substring(int beginIndex, int endIndex)
        //[begin, end)
        return s.substring(low + 1, high);
    }
}